### PR TITLE
Fix News dependencies and double Loading component issues.

### LINKED ISSUE #1: It has 2 loading spinners at the start
During the loading of the news for the first time, there appear 2 spinners at the same time. Feel free to take up this issue and work on it, if you face a problem in solving it we will help you with it

## Screenshot

Screenshot 2021-09-19 at 7 48 22 PM

### LINKED ISSUE #2: Console is showing a useEffect Hook error
The useEffect hook of React is showing an error on console.

## Screenshot
Screenshot 2021-09-19 at 8 00 40 PM

### PR DESCRIPTION
**Pull request by Ruberoni**
As discussed here [https://stackoverflow.com/a/60327893](https://stackoverflow.com/a/60327893), the workaround to Fix #2 (Linter problem) is to wrap the function [to be called in useEffect] in an useCallback.

The formatting was done by my editor, if it's a problem I could revert it.

Also, we could benefit from this Pull Request and commit changes to fix #1, let me know. That problem is due to using the Loader component inside `InfiniteScroll` and also using it outside, we have to decide a place, IMO it's better the one inside `InfiniteScroll` as its capabilities can do all the work and reduce code.

image
See both `Loader` components in lines 86 and 91.

### CODE REVIEW

**Comment by Guneetsinghtuli** [comment]
I think it will become better for me to review the code if you could just revert the formatting. So I can see the specific lines where you have made the changes as currently for every line it is showing me a change making it difficult to review, maybe next time we can work on the overall formatting.

File: src/components/News.js [hunk 1]
```diff
@@ -1,89 +1,104 @@
...
+      setProgress(val);
+    },
+    [setProgress]
+  );
+  
+  const updateNews = useCallback(async () => {
+    _setProgress(30);
+    setLoading(true);
+    const url = `https://newsapi.org/v2/top-headlines?q=${props.country}&apiKey=${props.apiKey}&page=1&pageSize=10&language=en&category=${props.category}`;
+    const data = await fetch(url);
```
Comment (Ruberoni) [inline src/components/News.js:30]: Added a bit of error handling for this fetch. Also below in the `fetchMoreData` function

File: src/components/News.js [hunk 2]
```diff
@@ -1,89 +1,104 @@
...
+  
+  const [data, setData] = useState([]);
+  const [currentPage, setPage] = useState(1);
+  const [page, settotalPage] = useState(1)
+  const [loading, setLoading] = useState(true)
+
+  const capitalise = (str) => {
+    return (str.slice(0, 1)).toUpperCase() + str.slice(1)
+  }
+
```
Comment (Ruberoni) [inline src/components/News.js:17]: Below, I had to get setProgress property, and wrap it in useCallback, so the function doesn't change.

File: src/components/News.js [hunk 3]
```diff
@@ -1,89 +1,104 @@
...
+  const { setProgress } = props
+  const _setProgress = useCallback(
+    (val) => {
```
Comment (Ruberoni) [inline src/components/News.js:19]: I had to get `setProgress` property, and wrap it in `useCallback`, so the function doesn't change.

File: src/components/News.js [hunk 4]
```diff
@@ -1,89 +1,104 @@
...
+  );
+  
+  const updateNews = useCallback(async () => {
+    _setProgress(30);
+    setLoading(true);
+    const url = `https://newsapi.org/v2/top-headlines?q=${props.country}&apiKey=${props.apiKey}&page=1&pageSize=10&language=en&category=${props.category}`;
+    const data = await fetch(url);
+    const { status, code, articles, totalResults } = await data.json();
+    if (status === "error") {
+      console.log("[Network error] code:", code);
```
Comment (Ruberoni) [inline src/components/News.js:31]: Added a bit of error handling for this fetch. Also below in the fetchMoreData function

**Comment by Guneetsinghtuli** [comment]
So you did amazing work and solved one of the issues but I am still getting the first issue #1

**Comment by Ruberoni** [comment]
Thank you man, I'm going to try to solve #1 and push a commit in this pull request

**Comment by Guneetsinghtuli** [comment]
Thank you for your contribution 🎉🎉

**Pull request merged**